streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -650,6 +650,25 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
   final def concat[R1 <: R, E1 >: E, A1 >: A](other: => ZStream[R1, E1, A1]): ZStream[R1, E1, A1] =
     ZStream(UIO.succeed(self), UIO(other)).flatMap(ZStream.unwrap)
 
+  /**
+   * Emits elements of this stream with a fixed delay in between, regardless of how long it
```

Comment:
Hmm, if I'm not mistaken, this will only delay the first element, as the ref is updated to `true` afterwards.

Reply:
This is how I understand it:
- when the state is false, we just evaluate the action and turn the state to true
- when the state is true, we wait `duration` before evaluating the action and keep the state to true

Reply:
Oh right!! Sorry, Boolean blindness :-)